springwebclient-benchmark/src/main/java/benchmark/springwebclient/Engine.java

```diff
@@ -0,0 +1,91 @@
+package benchmark.springwebclient;
+
+import com.ss.benchmark.httpclient.common.HttpClientEngine;
+import io.netty.channel.ChannelOption;
+import io.netty.handler.timeout.ReadTimeoutHandler;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.client.reactive.ReactorClientHttpConnector;
+import org.springframework.web.reactive.function.BodyInserters;
+import org.springframework.web.reactive.function.client.ClientResponse;
+import org.springframework.web.reactive.function.client.WebClient;
+import reactor.core.publisher.Mono;
+import reactor.netty.http.client.HttpClient;
+import reactor.netty.resources.ConnectionProvider;
+
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.TimeUnit;
+
+public class Engine implements HttpClientEngine {
+
+    private WebClient webClient;
+
+    @Override
+    public void createClient(String host, int port) {
+        String baseUrl = url(host, port);
+        webClient = WebClient
+                .builder()
+                .clientConnector(new ReactorClientHttpConnector(createReactorNettyClient()))
+                .baseUrl(baseUrl)
+                .build();
+    }
+
+    @Override
+    public String blockingGET(String path) {
+        return webClient
+                .get()
+                .uri(path)
+                .exchange()
+                .flatMap(res -> consumeResponse(res))
+                .doOnError(t -> System.err.println("Failed requesting server: " + t.getMessage()))
```

Comment:
Are we doing this elsewhere?  I'm assuming the blocking `get` is still going throw an exception?

Reply:
Can't both of these blocking calls be  `return nonBlockingXyz(path).block();` ?

Reply:
I was following the reactor-netty code to be consistent in style
https://github.com/sharathsrinivasa/httpclientbenchmark/blob/master/reactornetty-benchmark/src/main/java/benchmark/reactornetty/Engine.java#L46 
Should we change here also then?

Reply:
Looks lot cleaner now, will push the changes.

Reply:
TODO : Replace sysout with throwable...

Reply:
@crankydillo As we spoke, no code change here. Could you please mark this as resolved?